components/org.wso2.carbon.identity.conditional.auth.functions.user/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/user/Utils.java

```diff
@@ -120,6 +125,70 @@ public static String getUserIdClaimURI(String federatedIdpName, String tenantDom
             throws IdentityProviderManagementException {
 
         String userIdClaimURI = null;
+        ClaimMapping[] claimMappings = getIdPClaimMapping(federatedIdpName, tenantDomain);
+        if (claimMappings == null || claimMappings.length < 1) {
+            return null;
+        }
+        ClaimMapping userNameClaimMapping = Arrays.stream(claimMappings).filter(claimMapping ->
+                StringUtils.equals(USERNAME_LOCAL_CLAIM, claimMapping.getLocalClaim().getClaimUri()))
+                .findFirst()
+                .orElse(null);
+        if (userNameClaimMapping != null) {
+            userIdClaimURI = userNameClaimMapping.getRemoteClaim().getClaimUri();
+        }
+        return userIdClaimURI;
+    }
+
+    /**
+     * Resolve the userID claim URI from the IdP mapping or the default OIDC mapping.
+     *
+     * @param federatedIdpName federated IdP name.
+     * @param tenantDomain  tenant domain.
+     * @return userIDcClaimURI.
```

Comment:
```suggestion
     * @return user ID ClaimURI.
```

Reply:
Shall we rename the method to resolveUsername otherwise this is misleading right?